e2e/ansible/playbooks/hyperconverged/test-replica-node-affinity/replica-node-affinity.yaml

```diff
@@ -0,0 +1,193 @@
+         register: pv_name
+
+       - name: Copy replica_patch.yaml to master
+         copy:
+           src: "{{ patch_file }}"
+           dest: "{{ result_kube_home.stdout }}"
+ 
+       - name: Get the first node name where replica1 is scheduled
+         shell: kubectl get po --all-namespaces -o wide | grep "{{ pv_name.stdout }}" | grep rep | awk '{print $8}' | awk 'FNR == 1 {print}'
+         args:
+           executable: /bin/bash
+         register: node1
+     
+       - name: Get the second node name where replica1 is scheduled
+         shell: kubectl get po --all-namespaces -o wide | grep "{{ pv_name.stdout }}" | grep rep | awk '{print $8}' | awk 'FNR == 2 {print}'
+         args:
+           executable: /bin/bash
+         register: node2
+    
+       - name: Update first node name in replica_patch file
+         replace:
+           path: "{{ patch_file }}"
+           regexp: 'nodename_where_replica_pod_1_got_scheduled'
+           replace: '{{ node1.stdout }}'
+     
+       - name: Update second node name in replica_patch file
+         replace:
+           path: "{{ patch_file }}"
+           regexp: 'nodename_where_replica_pod_2_got_scheduled'
+           replace: '{{ node2.stdout }}'
+
+       - name: Obtain the deployment name 
+         shell: kubectl get deploy --all-namespaces |grep "{{ pv_name.stdout }}" | grep rep |awk '{print $2}'
+         args:
+           executable: /bin/bash
+         register: deploy_name
+   
+       - name: Update the replica deployment with node affinity property
+         shell: kubectl patch deployment "{{ deploy_name.stdout }}" -p "$(cat replica_patch.yaml)"
+         args:
+           executable: /bin/bash
+         register: patch_out
+         until: "'patched' in patch_out.stdout"
+         delay: 20
+         retries: 5
+     
+       - name: Get pod name
+         shell: kubectl get pods --all-namespaces -o wide | grep "{{ pv_name.stdout }}" | grep rep | grep "{{ node1.stdout }}" | awk '{print $2}'
+         args:
+           executable: /bin/bash
+         register: pod_name 
+
+       - name: Delete the replica on one of the nodes
+         shell: kubectl delete pod "{{ pod_name.stdout }}" --grace-period=0 --force
+         args:
+           executable: /bin/bash
+    
+       - name: Check the available replicas in deployment
+         shell: kubectl get deploy --all-namespaces | grep "{{ pv_name.stdout }}" | grep rep | awk '{print $6}'
+         args: 
+           executable: /bin/bash
+         register: available_pods
+         until: "'2' in available_pods.stdout"
+         delay: 30
+         retries: 5
+  
+       - name: Check if the replica is scheduled again
+         shell: kubectl get pods --all-namespaces -o wide | grep "{{ pv_name.stdout }}" | grep rep | grep Running | grep "{{ node1.stdout }}"
+         args:
+           executable: /bin/bash
+         register: result
+         until: "'Running' in result.stdout"
+         delay: 30
+         retries: 15
+
```

Comment:
Can we add a check to see that the replicas is scheduled only on the "desired" nodes.